pkg/console/controllers/oauthclients/oauthclients.go

```diff
@@ -118,22 +155,128 @@ func (c *oauthClientsController) sync(ctx context.Context, controllerContext fac
+	clientConfig := utilsub.GetOIDCClientConfig(authnConfig)
+	if clientConfig == nil {
+		c.authStatusHandler.WithCurrentOIDCClient("")
+		c.authStatusHandler.Unavailable("OIDCClientConfig", "no OIDC client found")
+		return nil
+	}
+
+	if len(clientConfig.ClientID) == 0 {
+		err := fmt.Errorf("no ID set on OIDC client")
+		c.statusHandler.AddConditions(status.HandleProgressingOrDegraded("OIDCClientConfig", "MissingID", err))
+		return c.statusHandler.FlushAndReturn(err)
+	}
+	c.authStatusHandler.WithCurrentOIDCClient(clientConfig.ClientID)
+
+	if len(clientConfig.ClientSecret.Name) == 0 {
+		c.authStatusHandler.Degraded("OIDCClientMissingSecret", "no client secret in the OIDC client config")
+		return nil
+	}
+
+	clientSecret, err := c.configNSSecretsLister.Secrets(api.OpenShiftConfigNamespace).Get(clientConfig.ClientSecret.Name)
+	if err != nil {
+		c.authStatusHandler.Degraded("OIDCClientSecretGet", err.Error())
+		return err
 	}
 
-	oauthErrReason, oauthErr := c.syncOAuthClient(ctx, clientSecret, consoleURL.String())
-	c.statusHandler.AddConditions(status.HandleProgressingOrDegraded("OAuthClientSync", oauthErrReason, oauthErr))
+	secret, err := c.targetNSSecretsLister.Secrets(api.TargetNamespace).Get(secretsub.Stub().Name)
+	expectedClientSecret := secretsub.GetSecretString(clientSecret)
+	if apierrors.IsNotFound(err) || secretsub.GetSecretString(secret) != expectedClientSecret {
+		secret, _, err = resourceapply.ApplySecret(ctx, c.secretsClient, controllerContext.Recorder(), secretsub.DefaultSecret(operatorConfig, expectedClientSecret))
+		if err != nil {
+			c.statusHandler.AddConditions(status.HandleProgressingOrDegraded("OIDCClientSecretSync", "FailedApply", err))
+			return err
+		}
+	}
+
+	if valid, msg, err := c.checkClientConfigStatus(authnConfig, secret); err != nil {
+		c.authStatusHandler.Degraded("DeploymentOIDCConfig", err.Error())
+		return err
+
+	} else if !valid {
+		c.authStatusHandler.Progressing("DeploymentOIDCConfig", msg)
+		return nil
+	}
+
+	c.authStatusHandler.Available("OIDCConfigAvailable", "")
+	return nil
+}
+
+// TODO add godoc
+func (c *oauthClientsController) checkClientConfigStatus(authnConfig *configv1.Authentication, clientSecret *corev1.Secret) (bool, string, error) {
+	depl, err := c.targetNSDeploymentsLister.Deployments(api.OpenShiftConsoleNamespace).Get(api.OpenShiftConsoleDeploymentName)
+	if err != nil {
+		return false, "", err
+	}
+
+	deplAvailableUpdated := deploymentsub.IsAvailableAndUpdated(depl)
+	if !deplAvailableUpdated {
+		return false, "deployment unavailable or outdated", nil
+	}
+
+	if clientSecret.GetResourceVersion() != depl.ObjectMeta.Annotations["console.openshift.io/oauth-secret-version"] {
+		return false, "client secret version not up to date in current deployment", nil
+	}
+
+	serverCAConfigName := authnConfig.Spec.OIDCProviders[0].Issuer.CertificateAuthority.Name
```

Comment:
Spec.OIDCProviders will be a singleton ?

Reply:
for the first implementation (and until upstream gets structured authentication config stable), yes, it will be